agent.go

```diff
@@ -169,7 +181,8 @@ type agent interface {
 	createContainer(pod *Pod, c *Container) error
 
 	// startContainer will tell the agent to start a container related to a Pod.
-	startContainer(pod Pod, c Container) error
+	// It returns the guest PID for the container init process.
+	startContainer(pod Pod, c Container) (guestProcess, error)
```

Comment:
This is an issue. I mean it won't be used since the shim is expected to be started when the container is created... That's the topic of this discussion https://github.com/kata-containers/agent/issues/38#issuecomment-353249666 and here is a possible solution https://github.com/kata-containers/agent/issues/38#issuecomment-353284358.
But in any case, I am not sure the guestProcess from the StartContainer will be used since this will be too late.